sasl_authenticate_request.go

```diff
@@ -0,0 +1,33 @@
+package sarama
+
+type SaslAuthenticateRequest struct {
+	SaslAuthBytes []byte
+}
+
+func (r *SaslAuthenticateRequest) encode(pe packetEncoder) error {
+	if err := pe.putBytes(r.SaslAuthBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *SaslAuthenticateRequest) decode(pd packetDecoder, version int16) (err error) {
+	if r.SaslAuthBytes, err = pd.getBytes(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *SaslAuthenticateRequest) key() int16 {
+	return 36
```

Comment:
`36` is a magic number, should be a named constant.

Reply:
I modeled this after the way the other API structs are defined. Shall I put the constant here or is there a central place where they should go? I'll also need to do this for the response struct.

Reply:
package level constant should be fine for this.